model/task_queue.go

```diff
@@ -372,24 +382,61 @@ func updateTaskQueue(distro string, taskQueue []TaskQueueItem, distroQueueInfo D
 	return errors.WithStack(err)
 }
 
-func ClearTaskQueue(distro string) error {
+func ClearTaskQueue(distroId string) error {
 	grip.Info(message.Fields{
 		"message": "clearing task queue",
-		"distro":  distro,
+		"distro":  distroId,
 	})
+
+	catcher := grip.NewBasicCatcher()
+
+	distroQueueInfo, err := GetDistroQueueInfo(distroId)
+	if err != nil {
+		catcher.Add(errors.Wrap(err, "error clearing task queue"))
+	}
+	clearQueueInfo(&distroQueueInfo)
+	err = clearTaskQueueCollection(distroId, TaskQueuesCollection, distroQueueInfo)
+	if err != nil {
+		catcher.Add(errors.Wrap(err, "error clearing task queue"))
+	}
+
+	distroQueueInfo, err = GetDistroQueueInfoAlias(distroId)
+	if err != nil {
+		catcher.Add(errors.Wrap(err, "error clearing task queue"))
+	}
+	clearQueueInfo(&distroQueueInfo)
+
+	err = clearTaskQueueCollection(distroId, TaskAliasQueuesCollection, distroQueueInfo)
+	if err != nil {
+		catcher.Add(errors.Wrap(err, "error clearing task queue"))
+	}
+	return catcher.Resolve()
+}
+
+// clearQueueInfo takes in a DistroQueueInfo and blanks out appropriate fields
+func clearQueueInfo(distroQueueInfo *DistroQueueInfo) {
```

Comment:
Typically, it's not desirable to use in-out parameters as a way to produce output from go functions (unless it's absolutely necessary, like for `json.Unmarshal`) - it's fine to return a copied version of the struct though.